app/lib/feed_manager.rb

```diff
@@ -22,18 +22,8 @@ def filter?(timeline_type, status, receiver)
   end
 
   def push(timeline_type, account, status)
-    timeline_key = key(timeline_type, account.id)
-
-    if status.reblog?
-      # If the original status is within 40 statuses from top, do not re-insert it into the feed
-      rank = redis.zrevrank(timeline_key, status.reblog_of_id)
-      return if !rank.nil? && rank < 40
-      redis.zadd(timeline_key, status.id, status.reblog_of_id)
-    else
-      redis.zadd(timeline_key, status.id, status.id)
-      trim(timeline_type, account.id)
-    end
-
+    redis.zadd(key(timeline_type, account.id), status.id, status.reblog? ? status.reblog_of_id : status.id)
```

Comment:
For some reason your branch is "up to date" yet this area is outdated - did you overwrite it?